tests/Microsoft.VisualStudio.ProjectSystem.Managed.UnitTests/Mocks/ConfiguredProjectFactory.cs

```diff
@@ -32,4 +33,6 @@ public static ConfiguredProject ImplementUnconfiguredProject(UnconfiguredProject
 
         return mock.Object;
     }
+
+    internal interface ITestConfiguredProjectImpl : ConfiguredProject, ConfiguredProject2;
```

Comment:
Rather than introducing a new interface, we can use Moq's built-in support for this.

https://github.com/devlooped/moq/wiki/Quickstart#advanced-features

From memory I think it's something like:

```c#
var mock = new Mock<ConfiguredProject>();
mock.Setup(...);
...
var mock2 = mock.As<ConfiguredProject2>();
mock2.Setup(c => c.EnsureProjectEvaluatedAsync()).Returns(Task.CompletedTask);

return mock.Object;
```

Reply:
Though creating a new interface was a pretty clever way to achieve the same thing. Use whichever you prefer. I don't think there's much material difference.

Reply:
Ah, thanks! I didn't realize that feature existed. Changed